src/pcre2_compile.c

```diff
@@ -3637,6 +3650,13 @@ while (ptr < ptrend)
       goto FAILED;
       }
 
+    /* Jump here from '(?[...])'. */
```

Comment:
while "technically" correct, wouldn't it make more sense to put this comment next to the label and after the assignment?

maybe also worth getting an assert to ensure  `class_mode_state` always has a valid value?

Reply:
Good point. I have moved the comment next to the label.

I'm not sure about adding an assertion. It wouldn't be useful, unless I added extra code to reset `class_mode_state` so it didn't have a valid value, after exiting a class! And that code is just as likely to go wrong during future maintenance.

I've added a comment however above the label, explaining what needs to be set before the jump. Ideally all our gotos would have documented descriptions of what variables must be set before the jump!